Libraries/LibWeb/Credentials/CredentialsContainer.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2024, Miguel Sacristán <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "CredentialsContainer.h"
+
+#include <AK/Vector.h>
+#include <LibWeb/Credentials/Credential.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Platform/EventLoopPlugin.h>
+
+namespace Web::Credentials {
+GC_DEFINE_ALLOCATOR(CredentialsContainer);
+
+WebIDL::ExceptionOr<GC::Ref<CredentialsContainer>> CredentialsContainer::construct_impl(JS::Realm& realm)
+{
+    return realm.create<CredentialsContainer>(realm);
+}
+
+// https://www.w3.org/TR/credential-management-1/#abstract-opdef-request-a-credential
+GC::Ref<WebIDL::Promise> CredentialsContainer::get(CredentialRequestOptions const& options)
+{
+    // 1. Let settings be the current settings object.
```

Comment:
generally if we are calling realm() all over the place, we put `auto& realm = this->realm();` at the top of the method to avoid a bunch of method calls all over the place.